megazords/ios-rust/Cargo.toml

```diff
@@ -20,4 +20,5 @@ autofill = { path = "../../components/autofill" }
 push = { path = "../../components/push" }
 tabs = { path = "../../components/tabs" }
 places = {path = "../../components/places" }
-places-ffi = { path = "../../components/places/ffi" }
\ No newline at end of file
+places-ffi = { path = "../../components/places/ffi" }
+glean-ffi = { path = "../../components/external/glean/glean-core/ffi" }
```

Comment:
I'm not familiar with how the whole glean ios situation works, I need to go get myself familiarized with it - but in here, we ship the version of glean that we have as the submodule with our AS, then in `rust-components-swift` we import `glean-swift` I'm getting the feeling that we should only do one or the other, but again I'm not familiar at all with how all of that works, but wanted to make sure that this sounds OK?

As far as I can tell from Slack, the requirement for iOS was to bundle them together into one binary to reduce the size, should we then make sure the `rust-components-swift` doesn't import `glean-swift`? and instead, uses this one binary? @travis79 @badboy

Reply:
That would actually alleviate my concerns in https://github.com/mozilla/rust-components-swift/pull/9, if we're shipping the glean rust code with the AS binary, we don't need to treat it differently than any other component (i.e it would be tied to our releases, but that's OK, because it used to be tied to our releases anyways with iOS). I don't know how close to the truth this is, so happy to be corrected!

Reply:
Original bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1661539
The one argument left is probably still "increased binary size and startup time reduction due additional load on dyld".

Given that focus-ios is fine with consuming it through rust-components-swift, maybe that isn't as big of a problem anymore? Or this was not specifically looked at at all.
So at the very least we would need to check how this affects firefox-ios. I don't know how to check that though (nor do I have the time to look into it)

Reply:
Yeah so this is definitely where I expected most of the discussion to take place. However we essentially are now doing what our Carthage build was doing (in megazords/ios) where we actually build glean and package it in the same shared libs. This does indeed mean we do not need to add it as a dependency anymore in `rust-components-swift` and can actually make it a target. I also do like this option more as it gives people the option of consuming glean as a separate target rather as a specific dependency of Nimbus (ex. only Glean + Logins or something)